mne/epochs.py

```diff
@@ -2701,21 +2852,47 @@ def _concatenate_epochs(epochs_list, with_data=True):
         selection = np.concatenate((selection, epochs.selection))
         drop_log.extend(epochs.drop_log)
         event_id.update(epochs.event_id)
+        metadata.append(epochs.metadata)
     events = np.concatenate(events, axis=0)
+
+    # Create metadata object (or make it None)
+    n_have = sum(this_meta is not None for this_meta in metadata)
+    if n_have == 0:
+        metadata = None
+    elif n_have != len(metadata):
+        raise ValueError('%d of %d epochs instances have metadata, either '
+                         'all or none must have metadata'
+                         % (n_have, len(metadata)))
+    else:
+        pd = _check_pandas_installed(strict=False)
+        if pd is not False:
+            metadata = pd.concat(metadata)
+        else:  # dict of dicts
+            metadatas = metadata
+            metadata = OrderedDict()
+            keys = set(metadatas[0].keys())
+            for this_meta in metadatas[1:]:
+                if not keys == set(this_meta.keys()):
+                    raise ValueError('All epochs.metadata must contain the '
+                                     'same keys')
+            for key in keys:
+                metadata[key] = OrderedDict(
+                    sum((list(this_meta[key].items())
+                         for this_meta in metadatas), list()))
```

Comment:
can you put this in a separate function called _concat_metadata ?

Reply:
actually if we're okay with resetting / ignoring the index (row labels) then there is a simpler route, I'll put it in